tasks/Task.cpp

```diff
@@ -13,37 +15,76 @@ Task::~Task()
 {
 }
 
+static RigidBodyState convertToRBS(PingStatus const& data)
+{
+    RigidBodyState rbs;
 
+    rbs.position = Eigen::Vector3d(data.response.acoustic_fix.position.east,
+        data.response.acoustic_fix.position.north,
+        -data.response.acoustic_fix.position.depth);
+    return rbs;
+}
 
 /// The following lines are template definitions for the various state machine
 // hooks defined by Orocos::RTT. See Task.hpp for more detailed
 // documentation about them.
-
 bool Task::configureHook()
 {
-    if (! TaskBase::configureHook())
+
+    std::unique_ptr<usbl_seatrac::Driver> driver(new Driver());
+    iodrivers_base::ConfigureGuard guard(this);
+    if (!_io_port.get().empty()) {
+        driver->openURI(_io_port.get());
+    }
+    setDriver(driver.get());
+
+    mDestinationId = _destination_id.get();
+    mMsgType = _msg_type.get();
+
+    if (!TaskBase::configureHook()) {
         return false;
+    }
+
+    mDriver = move(driver);
+    guard.commit();
     return true;
 }
+
 bool Task::startHook()
 {
-    if (! TaskBase::startHook())
+    if (!TaskBase::startHook()) {
         return false;
+    }
     return true;
 }
+
 void Task::updateHook()
 {
+    PingStatus status = mDriver->Ping(mDestinationId,  mMsgType);
+    if (status.flag == 1) {
```

Comment:
- you should be sending something if there is an error
- it would make sense to also output the full acoustic fix data structure on a port, since it gives a lot more diagnostic data.

Reply:
Can I create another outport with the PingStatus structure?

void Task::updateHook()
{
    PingStatus status = mDriver->Ping(mDestinationId, mMsgType);
    _ping_status.write(status);
    if (status.flag == 1) {
        auto rbs = convertToRBS(status);
        _pose.write(rbs);
    }
        
    TaskBase::updateHook();
}